.github/actions/javascript/contributorChecklist/contributorChecklist.js

```diff
@@ -0,0 +1,125 @@
+        - [x] I verified the JSDocs style guidelines (in [\`STYLE.md\`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
+    - [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
+    - [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
+    - [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like \`Avatar\`, I verified the components using \`Avatar\` are working as expected)
+    - [x] I verified all code is DRY (the PR does not include any logic written more than once, with the exception of tests)
+    - [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
+    - [x] If a new component is created I verified that:
+        - [x] A similar component does not exist in the codebase
+        - [x] All props are defined accurately and each prop has a \`/** comment above it */\`
+        - [x] Any functional components have the \`displayName\` property
+        - [x] The file is named correctly
+        - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
+        - [x] The only data being stored in the state is data necessary for rendering and nothing else
+        - [x] For Class Components, any internal methods passed to components event handlers are bound to \`this\` properly so there are no scoping issues (i.e. for \`onClick={this.submit}\` the method \`this.submit\` should be bound to \`this\` in the constructor)
+        - [x] Any internal methods bound to \`this\` are necessary to be bound (i.e. avoid \`this.submit = this.submit.bind(this);\` if \`this.submit\` is never passed to a component event handler like \`onClick\`)
+        - [x] All JSX used for rendering exists in the render method
+        - [x] The component has the minimum amount of code necessary for its purpose and it is
+    - [x] If a new CSS style is added I verified that:
+        - [x] A similar style doesn’t already exist
+        - [x] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. \`StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG\`)
+    - [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like \`Avatar\` is modified, I verified that \`Avatar\` is working as expected in all cases)
+    - [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.`;
```

Comment:
Hmm, I feel like this approach of essentially duplicating the checklist text here and searching for its presence via `includes` has the potential to create headaches. If the checklist text changes between the opening of the PR and the time it's being reviewed, the check could fail until the author/reviewer hunts down the latest version of the list and copy/pastes it in. For example there have been already updates to the checklist such as [this](https://github.com/Expensify/App/commit/32cfb97e5a8dee1db0be8207a84c5259d8ebcdee) since the opening of this PR. (I _suppose_ you could argue that's a feature not a bug, but 😐). Without digging into the workflow code a bit, it might not be obvious how to fix the failure.

Additionally, the duplication means the lists are prone to diverging (e.g. if someone updates `PULL_REQUEST_TEMPLATE.md` without updating this file). Would it be possible to read the contents of the MD file (or the PR body?), use a regex to isolate the checklist, then search for any unchecked boxes instead?

Reply:
> Without digging into the workflow code a bit, it might not be obvious how to fix the failure.

This is all true and fair feedback! I can update the error message, but will also update the tests with this new change from `main`.

> Would it be possible to read the contents of the MD file

Yes, it should be possible to read the PR checklist template as it's in the repo, but I do wonder if it's the direction we want to head in? We've been discussing how to improve this and have been discussing possibly having a bot comment with the checklist depending on the contents of the PR.

Additionally, I did initially try doing this via regex, and it's a very nasty regex.

Reply:
> having a bot comment with the checklist depending on the contents of the PR

Oh nice, that does sound like a fabulous alternative

> it's a very nasty regex

😬 Fair

Reply:
Yep, I'm on board